services/motion/builtin/move_request.go

```diff
@@ -229,16 +232,37 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 	if !ok {
 		return nil, resource.DependencyNotFoundError(movementSensorName)
 	}
+	before := time.Now()
+	// NOTE: This is the location of the robot, as repoorted by the GPS
+	// at the beginning of the MoveOnGlobe call.
 	origin, _, err := movementSensor.Position(ctx, nil)
+	ms.logger.Debugf("move on globe origin: %v", origin)
+	validatedExtra.extra["origin"] = origin
+	after := time.Now()
 	if err != nil {
 		return nil, err
 	}
 
+	if err := motion.LogGP(*origin, before, after); err != nil {
+		return nil, err
+	}
+
+	if err := motion.ValidateGeopoint(origin); err != nil {
+		return nil, err
+	}
+
 	// add an offset between the movement sensor and the base if it is applicable
+	// NOTE: Why do we create the baseOrigin as a pose in frame with a zero pose?
```

Comment:
TPspace motion problems are posed with everything relative to the base. Thus the base starts with the zero pose, and everything else is given its pose relative to that.

Reply:
Does that mean that the first step of the plan is always `be where you currently are`?

Reply:
Yes :)